### PR TITLE
The Fancyhat Fix

### PR DESCRIPTION
**Pull request by NPC1314**
## About The Pull Request

You know when I said I could fix the fancy hat in 20 seconds? Well turned out it needed a resprite, and I discovered Nitemaidens were a bit of a joke, and a couple more bugs, so fixed em all.

* Fancy hat now gets a onmob that actually sits on the head and not over your face
* Sinistar helmet icon path fixed so will be visible again
* a all races define just made a little clearer what it is
* new mood buffs for being clean added, tho not implemented beyond the herbal soap.
* herbal soup was  totally useless and broken, made a working version. Need to undress to get bathed, has messages, mood buffs, sounds etc.
* Nitemaiden get soap and cloth in their pouch for cleaning and swimming fixed to 2 since 1-2 random felt very pointless and they get more swimming done than most in town.
* Bandit lair had wrong area in a section, fixed to avoid incorrect teleports etc.
* Forces & Godless added to allowed Barbarian patrons, not meant to be strict one god templar role, more free to make your own take

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

File: code/modules/jobs/job_types/roguetown/adventurer/types/combat/barbarian.dm [hunk 1]
```diff
@@ -13,14 +13,14 @@
 	cmode_music = 'sound/music/combat_barbarian.ogg'
 
 /datum/outfit/job/roguetown/adventurer/barbarian
-	allowed_patrons = list(/datum/patron/divine/ravox, /datum/patron/inhumen/graggar)
+	allowed_patrons = list(/datum/patron/divine/ravox, /datum/patron/divine/abyssor, /datum/patron/divine/malum, /datum/patron/divine/necra, /datum/patron/divine/dendor, /datum/patron/godless, /datum/patron/inhumen/graggar)
```
Comment (BadAtThisGame302) [inline code/modules/jobs/job_types/roguetown/adventurer/types/combat/barbarian.dm:16]: I thought Malum was a Saint not a force?

File: code/modules/clothing/rogueclothes/storage.dm [hunk 2]
```diff
@@ -215,6 +215,13 @@
 	new /obj/item/ammo_casing/caseless/rogue/bullet(src)
 	new /obj/item/ammo_casing/caseless/rogue/bullet(src)
 
+
+//Nitemaiden pouch
+/obj/item/storage/belt/rogue/pouch/nitemaiden/PopulateContents()
+	new /obj/item/natural/cloth(src)
```
Comment (BadAtThisGame302) [inline code/modules/clothing/rogueclothes/storage.dm:221]: Maybe giving them some zennies?

Comment (NPC1314) [reply code/modules/clothing/rogueclothes/storage.dm:221]: why not, tho they should work for it the lazy bums

File: code/modules/jobs/job_types/roguetown/adventurer/types/combat/barbarian.dm (hunk 1 above)
Comment (NPC1314) [reply code/modules/jobs/job_types/roguetown/adventurer/types/combat/barbarian.dm:16]: eh youre right, an oversight when copy pasting

File: code/datums/status_effects/rogue/roguebuff.dm [hunk 3]
```diff
@@ -41,6 +41,25 @@
...
+
+/datum/status_effect/buff/clean_plus/on_apply()
+	. = ..()
+	if(iscarbon(owner))
+		var/mob/living/carbon/C = owner
+		C.add_stress(/datum/stressevent/clean_plus)
+
+/atom/movable/screen/alert/status_effect/buff/clean_plus
+	name = "Clean"
+	desc = "<span class='nicegreen'>I feel very refreshed.</span>\n"
```
Comment (BadAtThisGame302) [inline code/datums/status_effects/rogue/roguebuff.dm:59]: span_nicegreen

File: code/datums/stress/positive_events.dm [hunk 4]
```diff
@@ -128,6 +128,16 @@
 	stressadd = -5
 	desc = span_green("Down with the tyranny!")
 
+/datum/stressevent/clean
+	timer = 15 MINUTES
+	stressadd = -1
```
Comment (BadAtThisGame302) [inline code/datums/stress/positive_events.dm:133]: What's with the normal clean moodlet? Don't see it used anywhere.

Comment (NPC1314) [reply code/datums/stress/positive_events.dm:133]: its going to be used for some other stuff, felt weird to not add it while adding the + version.

**Review by Optimism333 — APPROVED** [review]
Fancy hats must be saved.

**Pull request merged**